complex.c

```diff
@@ -704,15 +704,12 @@ nucomp_s_polar(int argc, VALUE *argv, VALUE klass)
 {
     VALUE abs, arg;
 
-    switch (rb_scan_args(argc, argv, "11", &abs, &arg)) {
-      case 1:
-        nucomp_real_check(abs);
-        return nucomp_s_new_internal(klass, abs, ZERO);
-      default:
-        nucomp_real_check(abs);
+    if (rb_scan_args(argc, argv, "11", &abs, &arg) == 1) {
+        arg = ZERO;
+    } else {
         nucomp_real_check(arg);
-        break;
     }
+    nucomp_real_check(abs);
```

Comment:
The reason `nucomp_real_check(abs)` was written twice would be the order of the checks.